File: mili-api/src/main/java/org/leavesmc/leaves/event/BukkitEvent.java
package org.leavesmc.leaves.event;

import org.bukkit.event.Event;
import org.bukkit.event.HandlerList;

public class BukkitEvent extends Event {

    private static final HandlerList HANDLERS = new HandlerList();

    public static HandlerList getHandlerList() { return HANDLERS; }

    @Override
    public HandlerList getHandlers() { return HANDLERS; }
}

File: mili-api/src/main/kotlin/org/leavesmc/leaves/event/BukkitEvent.kt
@file:JvmName("BukkitEventKt")

package org.leavesmc.leaves.event

import org.bukkit.event.Event
import org.bukkit.event.HandlerList

/**
 * Base class for Kotlin Bukkit events with automatic HandlerList management.
 *
 * Kotlin data classes can extend this to eliminate the boilerplate
 * `HandlerList` + `getHandlerList` + `getHandlers` pattern.
 *
 * Usage:
 * ```kotlin
 * class BotCreateEvent(...) : BukkitEvent() {
 *     companion object {
 *         @JvmStatic val HANDLERS = HandlerList()
 *         @JvmStatic fun getHandlerList() = HANDLERS
 *     }
 *     override fun getHandlers() = HANDLERS
 * }
 * ```
 */
abstract class BukkitEvent : Event() {
    abstract override fun getHandlers(): HandlerList
}
